dali/test/python/test_operator_laplacian.py

```diff
@@ -0,0 +1,610 @@
+def _test_kernels(num_dims, smoothing, normalize):
+    batch_size = (max_window_size + 2 - min_window_size) // 2
+
+    def get_inputs():
+        ones = []
+        window_sizes = []
+        scales = []
+        padding = 2
+        for k in range(min_window_size, max_window_size + 2, 2):
+            a_size = k + padding
+            a = np.full((a_size,) * num_dims, 0, dtype=np.float32)
+            a[(a_size // 2,) * num_dims] = 1
+            ones.append(a)
+            if smoothing:
+                window_sizes.append(np.array(k, dtype=np.int32))
+                total_win_size = num_dims * k
+            else:
+                window_sizes.append(
+                    np.array([k, *[1] * (num_dims - 1)], dtype=np.int32))
+                total_win_size = k + num_dims - 1
+            exponent = total_win_size - 2 - num_dims
+            scales.append(np.array(2.**(-exponent), dtype=np.float32))
+        return ones, window_sizes, scales
+
+    @pipeline_def
+    def pipeline():
+        ones, window_sizes, scales = fn.external_source(
+            get_inputs, num_outputs=3)
+        kernels = fn.laplacian(
+            ones, window_size=window_sizes, dtype=types.FLOAT, normalize=normalize)
+        return kernels, scales
+
+    def outer(*vs):
+        acc = np.array([1.])
+        for v in vs:
+            acc = np.outer(acc, v)
+        return acc.reshape(tuple(len(v) for v in vs))
+
+    def get_cv2_kernel(k, smoothing):
+        d, s = cv2.getDerivKernels(2, 0, k)
+        if not smoothing:
+            s = np.zeros(k)
+            s[k // 2] = 1.
+        windows = [[d if i == j else s for j in range(
+            num_dims)] for i in range(num_dims)]
+        return sum(outer(*ws) for ws in windows)
+
+    pipe = pipeline(num_threads=4, batch_size=batch_size,
+                    device_id=types.CPU_ONLY_DEVICE_ID)
+    pipe.build()
+    (kernels, scales) = pipe.run()
+    kernels = [np.array(ker)[(slice(1, -1),) * num_dims] for ker in kernels]
+    scales = [np.array(sf).item() for sf in scales]
+    rng = range(min_window_size, max_window_size + 2, 2)
+    assert (len(kernels) == len(rng) == len(scales))
+    for scale_factor, kernel, k in zip(scales, kernels, rng):
+        baseline_kernel = get_cv2_kernel(k, smoothing)
+        if normalize:
+            baseline_kernel *= scale_factor
+        np.testing.assert_allclose(kernel, baseline_kernel, rtol=1e-5)
+
+
+def test_kernels():
+    for num_dims in [1, 2, 3]:
+        for normalize in [True, False]:
+            for smoothing in [True, False]:
+                yield _test_kernels, num_dims, smoothing, normalize
+
+
+@pipeline_def
+def laplacian_pipe(window_size, in_type, out_type, normalize, grayscale):
+    # use OpenCV convention - window size 1 implies deriv kernel of size 3 and no smoothing
+    window_size = window_size if window_size > 1 else [3, 1]
+    if out_type is None:
+        dtype_arg = {}
+    else:
+        dtype_arg = {'dtype': out_type}
```

Comment:
```suggestion
    if out_type is None:
        dtype_arg = {}
    else:
        dtype_arg = {'dtype': out_type}
```
No need for that. You can simply do `dtype=out_type`. If out_type is None, it'd be the same as not providing the arg at all.

Reply:
done